File: neetcode/ClimbingStairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int x=1,y=1;
        for(int i=2; i<=n; ++i) {
            int t=x;
            x+=y;
            y=t;
        }
        return x;
    }
};

File: Library/ClimbingStairs.java
import java.util.*;

class ClimbingStairs {
    public int climbStairs(int n) {
        if (n<=2) return n;

        int a=2, b=1, c=0;
        for (int i=3; i<n; i++) {
            c = a;
            a = a+b;
            b = c;
        }

        return a+b;
    }
}
